src/box/lua/config/init.lua

```diff
@@ -498,15 +498,30 @@ function methods.reload(self)
     end
 end
 
+function methods._hierarchy_info(self)
+    if self._configdata_applied == nil then
+        return {}
+    end
+
+    local names = self._configdata_applied:names()
+    return {
+        group = names.group_name,
+        replicaset = names.replicaset_name,
+        instance = names.instance_name,
+    }
```

Comment:
nit: right now `_config_applied:names()` will always return a table, is there a test, that future changes will never break this assumption? I'm mostly worried, because `_configdata_applied` is an internal structure and in future someone may change its behavior in a way that will break this code. We could just add something like `if names == nil then names = {} end` here and it shouldn't break ever)

Reply:
It would be dead code and it would create a false impression about the `:names()` contract.

An instance always have group/replicaset/instance names by construction. If `<configdata object>:names()` breaks this assumption, we have to change `applier/box_cfg.lua` and `applier/sharding.lua`, which also lean on it.

I would leave it as is.

Reply:
Thanks for the explanation!